Userland/Libraries/LibELF/DynamicLinker.cpp

```diff
@@ -781,6 +762,9 @@ Examples of static-pie ELF objects are ELF packers, and the system dynamic loade
 #endif
     }
 
+    // Manually end DependencyOrdering lifetime to free DynamicLoader objects.
+    objects.~DependencyOrdering();
```

Comment:
This is pretty unfortunate. could we store the objects in an optional and clear that optional instead? Calling destructors manually outside of data structure types makes me uncomfortable.